source/common/upstream/subset_lb.cc

```diff
@@ -110,6 +112,63 @@ void SubsetLoadBalancer::refreshSubsets(uint32_t priority) {
   update(priority, host_sets[priority]->hosts(), {});
 }
 
+void SubsetLoadBalancer::initSubsetSelectorMap() {
+  selectors_ = std::make_shared<SubsetSelectorMap>();
+  SubsetSelectorMapPtr selectors;
+  for (const auto& subset_selector : subset_selectors_) {
+    const auto& selector_keys = subset_selector->selector_keys_;
+    uint32_t pos = 0;
+    selectors = selectors_;
+    for (const auto& key : selector_keys) {
```

Comment:
What do you think about skipping this loop if `subset_selector->fallback_policy_` is undefined? As it stands, the default behavior will be to build up this data structure with all the subsets even if no subset has a custom fallback policy configured. If you skip adding the subsets with no custom fallback, then `SubsetSelectorMap.fallback_policy_` doesn't need to be optional and you can get rid of that complicated ternary below.

Reply:
agree, thanks for pointing out